0003-secret-seed-storage-and-key-derivation.md

```diff
@@ -0,0 +1,195 @@
+
+**In practice, No.**
+
+In Theory and in the case of Ethereum only you could scan the blockchain for any HTLC looking contract paying money to an address you generated deterministically from your seed and reclaim it.
+
+In Bitcoin you can't because you'll never be able to compute the P2WSH scriptPubKey without knowing the other person's input.
+If it has already been redeemed, you could scan for redeem transactions with script that matches our Bitcoin HTLC, recover the secret and then look for the HTLC on the other chain (as above, assuming it's Ethereum).
+
+I don't think this partial recovery is worth putting the significant effort that it would require to implement.
+
+### Should it be encrypted?
+
+I think this should be optional.
+A COMIT node doesn't own the funds long term.
+It can only control the funds of protocols that are in being executed.
+This is different to LND where if someone gets access to your seed they can take all your funds regardless of whether you have channels open or not.
+
+The best attack vector against an unencrypted seed is for the attacker to get short term access to the system COMIT is running on, covertly steal their secret seed and then wait for them to execute a swap and then redeem to your own address.
+Encrypting the seed does give you protection against this.
+
+Having said that, giving an attacker access to your COMIT node will probably give them access to the HTTP API which they can use to redeem funds to their own address.
+This is probably a better attack vector than the above.
+
+note: We could prevent this attack by giving the node a way of generating the final redeem addresses without getting them from the api.
+
+I think we should leave it unencrypted for now, make encryption an option later on and then perhaps make encryption the default as a last step.
+
+### Where to store it?
+
+Unlike lnd, I don't think we should store secret seed in the database file.
+This allows users to backup the secret seed separately from the database (although they need to backup both).
+Given that the two are linked, we should store a hash of the secret seed along with the swap id (in the same row or whatever).
+This allows the user to change the secret seed while not invalidating the entire database.
+
+By default I'd put it somewhere like:
+
+```
+~/.comit/secret_seed
+```
+
+But its location should be configurable with `secret_seed_file` in the `.toml` config.
+
+The format of the file should be toml with a single entry `secret_seed` (so we can `encrypted_secret_seed` later).
+
+We should make the permissions restrictive so only the user has read permission on it.
+
+#### What about encrypting the database?
+
+I don't think it's worth encrypting the database yet until we have a motivation to do it.
+An attacker stealing the database would learn what protocol executions you're involved in so atm it's just a privacy issue.
+
+We *could* encrypt the database we the `secret_seed` seeing as they're separated.
```

Comment:
Can you rephrase this sentence please (not sure if I understand it)?

Reply:
I think he wants to say:

```suggestion
We *could* encrypt the database with the `secret_seed` seeing as they're separated.
```